client-local/src/main/java/com/cloudera/livy/client/local/ContextLauncher.java

```diff
@@ -247,6 +247,10 @@ public void run() {
       }
       launcher.addAppArgs("--remote-host", serverAddress);
       launcher.addAppArgs("--remote-port",  serverPort);
+      if ("sparkr".equals(conf.get("session.kind"))) {
+        launcher.addSparkArg("--archives", conf.get("sparkr.package"));
```

Comment:
I'd rather see a `LocalConf.Entry` being created for this setting instead of hardcoding strings.

But more importantly, you shouldn't be using `--archives`; instead, you should be merging this with the `spark.yarn.dist.archives` config option which the user might be setting. This is done above for `SPARK_JARS_KEY` - you could refactor the code so that you can reuse it here.